apps/src/templates/sectionAssessments/MultipleChoiceOverviewTable.jsx

```diff
@@ -84,83 +70,64 @@ class MultipleChoiceOverviewTable extends Component {
-        property: 'percentAnsweredOptionOne',
-        header: {
-          label: commonMsg.answerOptionA(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-            width: 90,
-          }},
-        },
-        cell: {
-          format: answerOptionsOneFormatter,
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-            width: 90,
-          }}
-        }
+  getNotAnsweredColumn = () => (
+    {
+      header: {
+        label: commonMsg.notAnswered(),
+        props: {style: tableLayoutStyles.headerCell},
       },
-      {
-        property: 'percentAnsweredOptionTwo',
-        header: {
-          label: commonMsg.answerOptionB(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-            width: 90,
-          }},
-        },
-        cell: {
-          format: answerOptionsTwoFormatter,
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-            width: 90,
-          }}
-        }
+      cell: {
+        props: {style: tableLayoutStyles.cell},
+      }
+    }
+  );
+
+  getAnswerColumn = (index) => (
+    {
+      header: {
+        label: alphabetMapper[index],
+        props: {style: tableLayoutStyles.headerCell},
       },
-      {
-        property: 'notAnswered',
-        header: {
-          label: commonMsg.notAnswered(),
-          props: {
-            style: {
-            ...tableLayoutStyles.headerCell,
-            width: 90,
-          }},
-        },
-        cell: {
-          format: notAnsweredFormatter,
-          props: {
-            style: {
-            ...tableLayoutStyles.cell,
-            width: 90,
-          }}
-        }
+      cell: {
+        format: answerColumnsFormatter,
+        props: {style: tableLayoutStyles.cell},
+      }
+    }
+  );
+
+  getQuestionColumn = (sortable) => (
+    {
+      header: {
+        label: commonMsg.question(),
+        props: {style: tableLayoutStyles.headerCell},
+        transforms: [sortable],
       },
-    ];
+      cell: {
+        props: {style: tableLayoutStyles.cell},
+      }
+    }
+  );
+
+  getColumns = (sortable) => {
+    const maxAnswerChoicesLength = this.props.questionAnswerData.reduce((answersTotal, currentAnswerCount) => {
+      return Math.max(answersTotal, currentAnswerCount.answers.length);
+    }, 0);
+
+    let dataColumns = [];
+    let columns = this.getQuestionColumn(sortable) ;
+
+    dataColumns.push({property: 'question', ...columns});
+
+    for (let i = 0; i < maxAnswerChoicesLength; i++) {
+      let questionOption = this.getAnswerColumn(i);
+
+      dataColumns.push({property: 'percentAnswered' , ...questionOption});
+    }
+
+    // Add 2 to maxAnswerChoicesLength to ensure notAnswered is the last column.
+    // maxAnswerChoicesLength does not include question column.
+    dataColumns.push({property: 'notAnswered', ...this.getNotAnsweredColumn(maxAnswerChoicesLength + 2)});
```

Comment:
You no longer need to provide the argument to `getNotAnsweredColumn` since you removed it from the definition. The comment explaining it can go too.

Reply:
Makes sense!  Thanks.